src/ValueObjects/Messages/Support/Media.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Prism\Prism\ValueObjects\Messages\Support;
+
+use finfo;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Storage;
+use InvalidArgumentException;
+use Prism\Prism\Concerns\HasProviderMeta;
+
+class Media
+{
+    use HasProviderMeta;
+
+    protected ?string $localPath = null;
+
+    protected ?string $storagePath = null;
+
+    protected ?string $url = null;
+
+    protected ?string $rawContent = null;
+
+    protected ?string $base64 = null;
+
+    protected ?string $mimeType = null;
+
+    /**
+     * @deprecated Use `fromLocalPath` instead.
+     */
+    public static function fromPath(string $path): static
+    {
+        return self::fromLocalPath($path);
+    }
+
+    public static function fromLocalPath(string $path): static
+    {
+        if (! is_file($path)) {
+            throw new InvalidArgumentException("$path is not a file");
+        }
+
+        $content = file_get_contents($path);
+
+        if ($content === '' || $content === '0' || $content === false) {
+            throw new InvalidArgumentException("$path is empty");
+        }
+
+        $mimeType = File::mimeType($path);
+
+        if ($mimeType === false) {
+            throw new InvalidArgumentException("Could not determine mime type for {$path}");
+        }
+
+        $instance = new static;
+
+        $instance->localPath = $path;
+        $instance->rawContent = $content;
+        $instance->mimeType = $mimeType;
+
+        return $instance;
+    }
+
+    public static function fromStoragePath(string $path, ?string $disk = null): static
+    {
+        $disk = Storage::disk($disk);
+
+        if ($disk->exists($path) === false) {
+            throw new InvalidArgumentException("$path does not exist on the '$disk' disk");
+        }
+
+        $content = $disk->get($path);
+
+        if ($content === '' || $content === '0' || $content === false) {
+            throw new InvalidArgumentException("$path on the '$disk' disk is empty.");
+        }
+
+        $mimeType = (new finfo(FILEINFO_MIME_TYPE))->buffer($content);
```

Comment:
Laravel Storage has mimeType function, I think it would be appropriate to use it here. 
```suggestion
        $mimeType = $disk->mimeType($path);
```

Reply:
Ah, I thought it did but my IDE complained so thought I was going mad. Tested, and this does work, despite VSCode's red squiggle...